### PR TITLE
feat: ito unlucky swap

### PR DESCRIPTION
**Pull request by zhouhanseng**

### CODE REVIEW

File: packages/maskbook/src/web3/hooks/useTransactionState.ts [hunk 1]
```diff
@@ -15,6 +15,8 @@ export enum TransactionStateType {
     FAILED,
     /** Reject by external provider */
     REJECTED,
+    /** Workaround: ITO swapp fail due to unlucky*/
+    ITO_UNLUCKY,
```
Comment (guanbinrui) [inline packages/maskbook/src/web3/hooks/useTransactionState.ts:19]: If the unlucky status means the transaction failed, then you should use `TransactionType.FAILED`.

**Comment by guanbinrui** [comment]
@zhouhanseng Please rebase it onto `hotfix/1.32.1` and change the merge base to that branch.

File: packages/maskbook/src/web3/hooks/useTransactionState.ts (hunk 1 above)
Comment (zhouhanseng) [reply packages/maskbook/src/web3/hooks/useTransactionState.ts:19]: It is not FAILED, but Confirm.

File: packages/maskbook/src/plugins/Ethereum/UI/TransactionDialog.tsx [hunk 2]
```diff
@@ -140,6 +143,21 @@ function TransactionDialogUI(props: TransactionDialogUIProps) {
                         </Typography>
                     </>
                 ) : null}
+                {state.type === TransactionStateType.ITO_UNLUCKY ? (
```
[the remark this answers is not in the document]
Comment (zhouhanseng) [reply packages/maskbook/src/plugins/Ethereum/UI/TransactionDialog.tsx:146]: I will try to find a better way to do it in the future.

**Comment by Jack-Works** [comment]
do you sure? this PR looks so huge and seems like containing changes from the master branch cc @zhouhanseng @guanbinrui

**Comment by guanbinrui** [comment]
> are you sure? this PR looks so huge and seems like containing changes from the master branch cc @zhouhanseng @guanbinrui

Oh no...

**Comment by guanbinrui** [comment]
I've created another PR #3154. And the previous merge was revoked.

**Pull request merged**